components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -471,6 +474,22 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
                             "the database", e);
                 }
             }
+            // Check whether the authentication flow includes a SAML federated IdP and
+            // store the saml index with the session context key for the single logout.
+            if (context.getAuthenticationStepHistory() != null) {
```

Comment:
What will happen if context.getAuthenticationStepHistory() is null? can we add some debug logs for that

Reply:
Fixed with PR https://github.com/wso2/carbon-identity-framework/pull/3738